### ISSUE TITLE
How to call a single batch for ASR

### ISSUE THREAD

**Issue by ak3ra** [SunbirdAI/salt#13]
I am trying to figure out the leb module's method to load a single batch for training;

given:
```python
ds = leb.dataset.create(config)
```

What would be the equivalent method to do: 

```python
train_dataset = ds.map(
    prepare_dataset,
    batch_size=4,
    batched=True,
    num_proc=1,
)
```

where `train_dataset` contains audio and corresponding labels for a batch size of 4

This will eventually be documented in the docs in PR https://github.com/SunbirdAI/leb/pull/12

**Comment by jqug** [comment]
I've put an example in this [notebook](https://colab.research.google.com/drive/1RBXEyCPvsru4opZ5ZS5gBnu2lS71NvXc#scrollTo=UcjxkM11Y85V), hope that helps to illustrate it.

Basically it's nearly what you wrote, except that `ds` is of type `IterableDataset` (rather than the normal `Dataset`), meaning that it's lazy loaded ([more info](https://huggingface.co/docs/datasets/en/about_mapstyle_vs_iterable)). The `ds.map()` function is a little different and doesn't have a `num_proc` option.

Note that the dataset doesn't need to be batched in order for the training to run (I think). The `batch_size` and `batched` parameters in the mapping just refer to how many records are preprocessed with the `prepare_dataset()` function at once, and those params don't affect the resulting dataset. The preprocessing batch size and the batches used for training the model might be different.

**Comment by ak3ra** [comment]
Thanks! This is now clear. 
Closing this issue as resolved
